### ISSUE TITLE
[FEATURE] Add Experimental badges and messaging

### ISSUE THREAD

**Issue by kamingleung** [opensearch-project/dashboards-flow-framework#461]
**Is your feature request related to a problem?**
We will first offer this feature as an Experimental Feature to gather users feedback and improvements. We will need to visually inform users this is Experimental and not ready for production use. We should offer a way for users to leave feedback as they try out the feature.

**What solution would you like?**
We will display an Experimental badge in selected areas, when users click on the badge, a popover will show a message:
`OpenSearch Flow is experimental and should not be used in a production environment.`. Clicking on the button `Provide feedback on Github` will open a new tab to [this page](https://github.com/opensearch-project/dashboards-flow-framework/issues/new/choose), so users can provide feedback by creating an issue.
![image](https://github.com/user-attachments/assets/5fde1c9a-cb24-4414-aa58-e9e44e37d149)

**Display the badge AND popover in the following 3 areas:**
For the badge, use [OuiBetaBadge](https://oui.opensearch.org/1.17/#/display/badge#beta-badge-type) with the beaker icon
1. Search overview page (place badge on the OpenSearch Flow tile)
![image](https://github.com/user-attachments/assets/99bf7650-c746-4ea7-9684-a40420107b3b)

2. Project list/create page (place badge next to the header)
![image](https://github.com/user-attachments/assets/8ad7e5a5-152a-49bd-8120-d4d36c60db1b)

3. Inside a workflow (place badge next to the action buttons)
![image](https://github.com/user-attachments/assets/48e03b52-4baa-4252-b72a-b8d2520924e2)

**Display only the beaker icon in the navigation (no badge component)**
![image](https://github.com/user-attachments/assets/5b61a8f7-52e6-4a14-b8ec-92efe13de933)
When users hover over the OpenSearch Flow item, we should update the native browser label to 
`OpenSearch Flow – Experimental feature`
![image](https://github.com/user-attachments/assets/3e66c270-c58c-4380-9199-98c628760692)

**Comment by ohltyler** [comment]
I've encountered several issues with core OSD flexibility to enable these flags in many different places:
1. List/header page (`useNewHomePage: true` ONLY). The new UI header framework does not allow for an extra icon or React components to the right of the actions list.
2. Inside a workflow page (`useNewHomePage: true` ONLY). The new UI header framework does not allow for an extra icon or React components to the right of the actions list.
3. In the navigation bar (both new AND old home page layouts). The navigation framework does not allow for icons to be visible, let alone extra functionality such as hovering.

It looks like all of these will require some sort of framework enhancements in core OSD & the owners of the new header layouts to expose this sort of functionality.

To recap:

- [ ] overview page complete
- [ ] workflows page (old UI) complete
- [ ] workflows page (new UI) not possible
- [ ] workflow details page (old UI) not complete - we can add this next to the workflow title if we want?
- [ ] workflow details page (new UI) not possible
- [ ] navigation (old UI) not possible
- [ ] navigation (new UI) not possible

@kamingleung  In the meantime I will open a PR with the changes possible to make today. Lmk your thoughts/priority on the other changes. Perhaps we can find a middleground and expose some labels somewhere else on the new UI, and I can open tracking issues in core OSD to address the gaps mentioned.

**Comment by kamingleung** [comment]
@AMoo-Miki 
Do we supporting adding badging to the new page header, like this?
![image](https://github.com/user-attachments/assets/8ad7e5a5-152a-49bd-8120-d4d36c60db1b)

Are we able to add badging to the new Application Header? Perhaps we can add sections similar to Visualize?
Proposed Fractal Application Header with Experimental Badge:
![image](https://github.com/user-attachments/assets/46d8cde5-092c-4c17-982c-2d84589bd9bc)

The new Visualize:
![image](https://github.com/user-attachments/assets/d88ede4a-093a-4b47-a74a-4f9e03feffa6)


cc. @lauralexis @ virajsanghvi – We may want to consider supporting similar patterns within our OUI components in the future.

**Comment by kgcreative** [comment]
I think we need to add a prop to page header for `isExperimental=true` which allows you to pass the right values.

**Comment by ohltyler** [comment]
Latest update:

- [x]  overview page complete https://github.com/opensearch-project/dashboards-flow-framework/pull/472
- [x]  workflows page (old UI) complete https://github.com/opensearch-project/dashboards-flow-framework/pull/472
- [x]  workflows page (new UI) complete #496 
- [x]  workflow details page (old UI) complete #485 
- [x]  workflow details page (new UI) complete #485 
- [ ]  navigation (old UI) icon + tooltips not possible
- [ ]  navigation (new UI) icon + tooltips not possible

**Comment by ohltyler** [comment]
As discussed, navigation is not possible with the current framework. Please reopen or add a standalone issue to close on this particular part.

**Comment by kamingleung** [comment]
> As discussed, navigation is not possible with the current framework. Please reopen or add a standalone issue to close on this particular part.

Got it. LGTM on the latest updates. This is a temporary solution. We will open a new issue once we have the components with props to support experimental badges.
